test/WebJobs.Script.Tests.Shared/TestChangeTokenSource.cs

```diff
@@ -10,26 +11,41 @@ namespace Microsoft.Azure.WebJobs.Script.Tests
 {
     public class TestChangeTokenSource<T> : IOptionsChangeTokenSource<T>
     {
-        private IChangeToken _changeToken;
-        private CancellationTokenSource _cts = new CancellationTokenSource();
+        private TestChangeToken _changeToken;
 
         public TestChangeTokenSource()
         {
-            _changeToken = new CancellationChangeToken(_cts.Token);
+            _changeToken = new TestChangeToken();
         }
 
         public string Name { get; set; }
 
         public void SignalChange()
         {
-            Interlocked.Exchange(ref _changeToken, NullChangeToken.Singleton);
-            _cts.Cancel();
-            _cts.Dispose();
+            var token = Interlocked.Exchange(ref _changeToken, new TestChangeToken());
+            token.SignalChange();
```

Comment:
Minor change to allow us to signal a change multiple times on the same token source.